src/chrome/logging/methodsCalledLogger.ts

```diff
@@ -59,38 +58,48 @@ export class MethodsCalledLogger<T extends object> {
 
     public wrapped(): T {
         const handler = {
-            get: <K extends keyof T>(target: T, propertyKey: K, _receiver: any) => {
-                const originalPropertyValue: any = target[propertyKey];
+            get: <K extends keyof T>(target: T, propertyKey: K, receiver: any) => {
+                const originalPropertyValue = target[propertyKey];
+                // tslint:disable-next-line: strict-type-predicates
                 if (typeof originalPropertyValue === 'function') {
                     return (...args: any) => {
+                        const callId = this.generateCallId();
                         try {
                             if (propertyKey === 'on' && args.length >= 2) {
                                 let listenerPossiblyWrapped = args[1];
                                 this._configuration.decideWhetherToWrapEventEmitterListener(this._objectToWrapName, propertyKey, args, name => listenerPossiblyWrapped = new MethodsCalledLogger(this._configuration, args[1], name).wrapped());
                                 args[1] = listenerPossiblyWrapped;
                             }
 
+                            this.logCallStart(propertyKey, args, callId);
                             const result = originalPropertyValue.apply(target, args);
                             if (!result || !result.then) {
```

Comment:
Does this pass the boolean expressions lint rule?

Reply:
What do you mean?

Reply:
Didn't we just enable the `strict-boolean-expressions` lint rule that ended up requiring lots of `isDefined` and similar calls? But here we're just testing `result` the normal "javascripty" way. Not a problem, just curious.

Reply:
You are right, I missed that.
result is of type any, so it doesn't trigger that rule, so tslint is not complaining...